coalib/output/ConsoleInteraction.py

```diff
@@ -855,38 +855,58 @@ def show_enumeration(console_printer,
     console_printer.print()
 
 
+def check_deprecation(show_description,
+                      show_params):
+    """
+    Shows a deprecation warning message if the parameters
+    show_description or show_params are not None.
+
+    :param show_description: This parameter is being checked
+                             for deprecation.
+    :param show_params:      This parameter is being checked
+                             for deprecation.
+    """
+    param_dict = {'show_description': show_description,
+                  'show_params': show_params}
+    for param_name, param_value in param_dict.items():
+        if param_value is not None:
+            logging.warning('{} parameter is deprecated'.format(param_name))
+
+
 def show_bear(bear,
-              show_description,
-              show_params,
               console_printer,
-              args=None):
+              args=None,
+              show_description=None,
+              show_params=None):
     """
     Displays all information about a bear.
 
     :param bear:             The bear to be displayed.
-    :param show_description: True if the main description should be shown.
-    :param show_params:      True if the details should be shown.
     :param console_printer:  Object to print messages on the console.
     :param args:             Args passed to coala command.
+    :param show_description: This parameter is deprecated.
+    :param show_params:      This parameter is deprecated.
     """
     console_printer.print(bear.name, color='blue')
 
     metadata = bear.get_metadata()
 
-    if show_description:
```

Comment:
a deprecated parameter is still valid, and must still be used, otherwise it isnt [deprecation](https://en.wikipedia.org/wiki/Deprecation).

if you are discontinuing use of an argument, but keeping it in the arg list, it is 'unused' or 'ignored', and any use of it is a error or a very visible warnings.  This is sometimes done because removing the arg breaks shit, and you cant give nice error message if everything is broken.  Hence, you leave the arg in there, and nicely inform the developer/user that they now *must* update the code.